src/query/graphite/native/builtin_functions.go

```diff
@@ -242,6 +243,42 @@ func timeShift(
 	}, nil
 }
 
+
+func timeSlice(ctx *common.Context, inputPath singlePathSpec, start string, end string) (ts.SeriesList, error) {
+	tzOffsetForAbsoluteTime := time.Duration(0)
+	startTime, err := graphite.ParseTime(start, time.Now(), tzOffsetForAbsoluteTime)
+	endTime, err := graphite.ParseTime(end, time.Now(), tzOffsetForAbsoluteTime)
+
+	if (err != nil) {
+		return ts.NewSeriesList(), err
+	}
+
+	input := ts.SeriesList(inputPath)
+	output := make([]*ts.Series, input.Len())
+
+	for i, series := range input.Values {
+		nanoSecondsPerStep := series.MillisPerStep() * 1000000
+		stepDuration := time.Duration(nanoSecondsPerStep)
```

Comment:
Use the time constants here, i.e.:
```suggestion
		stepDuration := time.Duration(series.MillisPerStep()) * time.Millisecond
```